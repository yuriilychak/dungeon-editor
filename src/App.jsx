import React, { Component } from 'react';
import './App.css';

import { withTranslation } from "react-i18next";

class App extends Component {

  render() {
      const { t } = this.props;

      return (
          <div className="App">
              <h2>{t('Welcome to React')}</h2>
          </div>
      );
  }
}

export default withTranslation()(App);
