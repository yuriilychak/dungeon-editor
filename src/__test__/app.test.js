import { createShallow, createMount } from "@material-ui/core/test-utils";
import React from "react";

import App from "../app";

jest.mock("../loader", () => ({
    loadStatic: jest.fn(() => Promise.resolve())
}));

jest.mock("../store", () => () => ({}));

jest.mock("../bottom-menu", () => ({
    BottomMenu: () => <span>BottomMenu</span>
}));

jest.mock("../top-menu", () => ({
    TopMenu: () => <span>TopMenu</span>
}));

jest.mock("../library", () => ({
    Library: () => <span>Library</span>
}));

jest.mock("../properties", () => ({
    Properties: () => <span>Properties</span>
}));

jest.mock("../new-project-dialog", () => ({
    NewProjectDialog: () => <span>NewProjectDialog</span>
}));

jest.mock("../new-file-dialog", () => ({
    NewFileDialog: () => <span>NewFileDialog</span>
}));

jest.mock("../export-project-dialog", () => ({
    ExportProjectDialog: () => <span>ExportProjectDialog</span>
}));

jest.mock("../rename-file-dialog", () => ({
    RenameFileDialog: () => <span>RenameFileDialog</span>
}));

jest.mock("../working-area", () => ({
    WorkingArea: () => <span>WorkingArea</span>
}));

describe("GIVEN App component", () => {
    it("WHEN render App with empty store THEN rendered result should match snapshot", () => {
        const wrapper = createShallow()(<App/>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("WHEN render App with non empty store THEN rendered result should match snapshot", () => {
        jest.spyOn(React, "useState").mockImplementation(() => [{ getState: jest.fn() }, jest.fn()]);
        jest.spyOn(React, "useEffect").mockImplementation(callback => callback());
        const wrapper = createMount()(<App/>);
        expect(wrapper.html()).toMatchSnapshot();
    });
});