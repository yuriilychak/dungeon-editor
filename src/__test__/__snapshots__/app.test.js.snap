// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN App component WHEN render App with empty store THEN rendered result should match snapshot 1`] = `"<div class=\\"loader\\"><div class=\\"loader-ring\\"><div></div><div></div><div></div><div></div></div></div>"`;

exports[`GIVEN App component WHEN render App with non empty store THEN rendered result should match snapshot 1`] = `"<div class=\\"loader\\"><div class=\\"loader-ring\\"><div></div><div></div><div></div><div></div></div></div>"`;
